RichText.cpp

```diff
@@ -32,6 +67,48 @@ void RichText::Line::setFont(const sf::Font &font)
 }
 
 
+////////////////////////////////////////////////////////////////////////////////
+std::size_t RichText::Line::getLength() const
+{
+    std::size_t count = 0;
+    for(sf::Text &text : m_texts)
+    {
+        count += text.getString().getSize();
+    }
+    return count;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+sf::Color RichText::Line::getCharacterColor(std::size_t pos) const
+{
+    assert(pos < getLength());
+    assert(pos >= 0);
+    return m_texts[convertLinePosToLocal(pos)].getColor();
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+sf::Uint32 RichText::Line::getCharacterStyle(std::size_t pos) const
+{
+    assert(pos < getLength());
+    assert(pos >= 0);
+    return m_texts[convertLinePosToLocal(pos)].getStyle();
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+sf::Uint32 RichText::Line::getCharacter(std::size_t pos) const
+{
+    assert(pos < getLength());
+    assert(pos >= 0);
+    // Similar to setCharacter()
+    sf::Text& text = m_texts[convertLinePosToLocal(pos)];
+    sf::String string = text.getString();
```

Comment:
Avoid a copy if you just want to access a character. So either a `sf::String&` or directly `text.getString()[pos]`.

Reply:
Yes! 👍